lib/views/movie_list_view.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+import '../repositories/movie_repository.dart';
+import '../text_styles/custom_text_style.dart';
+import '../ui_constants.dart';
+import 'movie_details_view.dart';
+import '../entities/movie.dart';
+
+class MovieListView extends StatefulWidget {
+  const MovieListView({super.key});
+
+  @override
+  State<MovieListView> createState() => _MovieListViewState();
+}
+
+class _MovieListViewState extends State<MovieListView> {
+  final MovieRepository movieRepository = MovieRepository();
+  static const double dividerHeight = 20;
+  static const String errorMessage = 'An error has occurred while loading data';
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      builder: (BuildContext context, AsyncSnapshot<List<Movie>> snapshot) {
+        if (snapshot.hasData) {
+          List<Movie>? movies = snapshot.data;
+          return ListView.separated(
+            itemCount: movies!.length,
+            itemBuilder: (context, index) {
+              return GestureDetector(
+                onTap: () {
+                  Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                      builder: (context) => MovieDetailsVIew(
+                        movie: movies[index],
+                      ),
+                    ),
+                  );
+                },
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    Text(
+                      movies[index].title,
+                      textAlign: TextAlign.center,
+                      style: CustomTextStyle.overviewTextStyleBoldWhite,
+                    ),
+                    Padding(
+                      padding: const EdgeInsets.all(MovieDetailsUiConstants.releaseDateRowPadding),
+                      child: Image.network(
+                        '${MovieDetailsUiConstants.imagesRouteBeginning}${movies[index].posterPath}',
+                      ),
+                    ),
+                  ],
+                ),
+              );
+            },
+            separatorBuilder: (BuildContext context, int index) {
+              return const Divider(
+                height: dividerHeight,
+              );
+            },
+          );
+        }
+        return const Text(
+          errorMessage,
+        );
+      },
+      future: movieRepository.getMovies(),
```

Comment:
We can add some Flutter best practices here. Please read this:
https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html

> If the future is created at the same time as the FutureBuilder, then every time the FutureBuilder's parent is rebuilt, the asynchronous task will be restarted. A general guideline is to assume that every build method could get called every frame, and to treat omitted calls as an optimization.

TLDR, you could make this widget Stateful, declare a Future<T> movies at the beginning, init that future in the initState method, and use that future in the FutureBuilder. Like here:
https://stackoverflow.com/a/68966640

Reply:
i took my time and i read it! and it's done.